lib/internal/bootstrap_node.js

```diff
@@ -697,4 +700,36 @@
   };
 
   startup();
+
+  function makePrimordials() {
+    const ReflectApply = Reflect.apply;
+
+    // This function is borrowed from the function with the same name on V8
+    // Extras' `utils` object. V8 implements Reflect.apply very efficiently in
+    // conjunction with the spread syntax, such that no additional special case
+    // is needed for function calls w/o arguments.
+    // Refs: https://git.io/vAOyO
+    function uncurryThis(func) {
+      return (thisArg, ...args) => ReflectApply(func, thisArg, args);
+    }
```

Comment:
@hashseed @bmeurer @devsnek can you confirm that this has _zero overhead_? Some of these calls sits in hot paths and I do not want to add an utility that we need to be careful in using.

Reply:
I see a 17% performance regression with this benchmark on Node 8, but with newest V8 there is none.

```javascript
const ReflectApply = Reflect.apply;

function uncurryThis(func) {
  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
}
var StringStartsWith = uncurryThis(String.prototype.startsWith);

var s = "Hi Matteo";
function test1() {
  var r = 0;
  console.time("monkey-patchable");
  for (var i = 0; i < 1E7; i++) {
    if (s.startsWith(i)) r++;
  }
  console.timeEnd("monkey-patchable");
  return r;
}

function test2() {
  var r = 0;
  console.time("cached original");
  for (var i = 0; i < 1E7; i++) {
    if (StringStartsWith(s, i)) r++;
  }
  console.timeEnd("cached original");
  return r;
}

test1();
test2();
```

Reply:
wow! Good to know!
Which version of V8 did you check? what we have on master? What we plan to have for Node 10?

Reply:
Also ran with current Node master. There is no regression there either.